app/collectors/launchTemplates.scala

```diff
@@ -0,0 +1,122 @@
+import software.amazon.awssdk.services.ec2.model.{
+  DescribeLaunchTemplateVersionsRequest,
+  LaunchTemplateVersion => AwsLaunchTemplateVersion
+}
+import utils.Logging
+
+import scala.jdk.CollectionConverters._
+import scala.language.postfixOps
+import scala.util.Try
+
+class LaunchTemplateCollectorSet(accounts: Accounts)
+    extends CollectorSet[LaunchTemplateVersion](
+      ResourceType("launch-templates"),
+      accounts,
+      Some(Regional)
+    ) {
+  val lookupCollector
+      : PartialFunction[Origin, Collector[LaunchTemplateVersion]] = {
+    case amazon: AmazonOrigin =>
+      AWSLaunchTemplateCollector(
+        amazon,
+        resource,
+        amazon.crawlRate(resource.name)
+      )
+  }
+}
+
+case class AsgLaunchTemplate(id: String, version: String)
+
+case class AWSLaunchTemplateCollector(
+    origin: AmazonOrigin,
+    resource: ResourceType,
+    crawlRate: CrawlRate
+) extends Collector[LaunchTemplateVersion]
+    with Logging {
+
+  val asgClient: AutoScalingClient = AutoScalingClient.builder
+    .credentialsProvider(origin.credentials.provider)
+    .region(origin.awsRegionV2)
+    .overrideConfiguration(AWS.clientConfig)
+    .build
+
+  val ec2Client: Ec2Client = Ec2Client.builder
+    .credentialsProvider(origin.credentials.provider)
+    .region(origin.awsRegionV2)
+    .overrideConfiguration(AWS.clientConfig)
+    .build
+
+  def crawl: Iterable[LaunchTemplateVersion] = {
+    val asgs =
+      asgClient.describeAutoScalingGroups().autoScalingGroups().asScala.toList
+    val launchTemplates = asgs
+      // not all asgs have launch templates (they might have launch configurations instead
+      .filter(asg => asg.launchTemplate() != null)
+      .map(asg =>
+        AsgLaunchTemplate(
+          asg.launchTemplate().launchTemplateId(),
+          asg.launchTemplate().version()
+        )
+      )
+
+    launchTemplates.map { lt =>
+      // might need to filter out asgs which don't have launch templates?
+      val requestLt = DescribeLaunchTemplateVersionsRequest.builder
+        .launchTemplateId(lt.id)
+        .versions(lt.version)
+        .build
+      val template = ec2Client
+        .describeLaunchTemplateVersions(requestLt)
+        .launchTemplateVersions()
+        .asScala
+        .head
+      LaunchTemplateVersion.fromApiData(template, origin)
+    }
+
+  }
+}
+
+object LaunchTemplateVersion {
+  def fromApiData(
+      config: AwsLaunchTemplateVersion,
+      origin: AmazonOrigin
+  ): LaunchTemplateVersion = {
+    LaunchTemplateVersion(
+      arn = config.launchTemplateId(),
+      name = config.launchTemplateName(),
+      imageId = config.launchTemplateData().imageId(),
+      region = origin.region,
+      instanceProfile =
+        Option(config.launchTemplateData().iamInstanceProfile().arn()),
+      createdTime = Try(config.createTime()).toOption,
+      instanceType = config.launchTemplateData().instanceType().toString,
+      securityGroups = Option(
```

Comment:
It looks like these are all empty, is that expected?

```sh
curl --silent https://prism.code.dev-gutools.co.uk/launch-templates | \
jq ".data.\"asg-launch-templates\"[].securityGroups" | \
uniq
```

Reply:
Sounds like a bug to me! Will have a look

Reply:
Resolved in https://github.com/guardian/prism/pull/918/commits/10ffc57a7094df7cb88e54b52b88e9bdfa57bc25